azuread/data_user_test.go

```diff
@@ -0,0 +1,44 @@
+package azuread
+
+import (
+	"fmt"
+	"math/rand"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceAzureADUser_byUserPrincipalName(t *testing.T) {
+	dataSourceName := "data.azuread_user.test"
+	id := strconv.Itoa(rand.Intn(9999))
```

Comment:
Could we change this to: 
```suggestion
	id := acctest.RandString(7)
```